src/server_manager/install_scripts/install_server.sh

```diff
@@ -185,44 +159,103 @@ install_shadowbox() {
     -e "SB_DEFAULT_SERVER_NAME=${SB_DEFAULT_SERVER_NAME:-}"
   )
   docker run -d "${docker_shadowbox_flags[@]}" "${SB_IMAGE}" >/dev/null
+}
 
-  log_step "Starting Shadowbox"
-
-  # TODO(dborkan): if the script fails after docker run, it will continue to fail
-  # as the names shadowbox and watchtower will already be in use.  Consider
-  # deleting the container in the case of failure (e.g. using a trap, or
-  # deleting existing containers on each run).
-
+function start_watchtower() {
   # Start watchtower to automatically fetch docker image updates.
-  # TODO(fortuna): Don't wait for Shadowbox to run this.
+  # Set watchtower to refresh every 30 seconds if a custom SB_IMAGE is used (for
+  # testing).  Otherwise refresh every hour.
+  readonly WATCHTOWER_REFRESH_SECONDS=$([ $SB_IMAGE ] && echo 30 || echo 3600)
   declare -a docker_watchtower_flags=(--name watchtower --restart=always)
   docker_watchtower_flags+=(-v /var/run/docker.sock:/var/run/docker.sock)
   docker run -d "${docker_watchtower_flags[@]}" v2tec/watchtower --cleanup --tlsverify --interval $WATCHTOWER_REFRESH_SECONDS >/dev/null
+}
+
+# Waits for Shadowbox to be up and healthy
+function wait_shadowbox() {
+  until curl --insecure -s "${LOCAL_API_URL}/access-keys" >/dev/null; do sleep 1; done
+}
+
+function create_first_user() {
+  curl --insecure -X POST -s "${LOCAL_API_URL}/access-keys" >/dev/null
+}
 
-  log_step "Starting Watchtower"
+function output_config() {
+  echo "$@" >> $ACCESS_CONFIG
+}
 
-  readonly SB_API_URL="https://${SB_PUBLIC_IP}:${SB_API_PORT}/${SB_API_PREFIX}"
-  # Wait for server to be ready
-  until curl --cacert "${SB_CERTIFICATE_FILE}" -s "${SB_API_URL}/access-keys" >/dev/null; do sleep 1; done
-  # Create a new user
-  curl --cacert "${SB_CERTIFICATE_FILE}" -X POST -s "${SB_API_URL}/access-keys" >/dev/null
+function add_api_url_to_config() {
+  output_config "apiUrl:${PUBLIC_API_URL}"
+}
 
-  log_step "Creating first user"
+function check_firewall() {
```

Comment:
I would expect most firewalls to allow connections from localhost addresses, even if they're blocked from the outside. This will have a lot of false positives - but I guess as a placeholder it's okay.

Reply:
@dborkan had the excellent point that he's seen the **converse**: a firewall that allows connections  from localhost but not externally. That would be worse than what we have - however, seems like you've covered both sides here by just providing a warning if we suspect the user will have problems.